src/zkscript/util/utility_scripts.py

```diff
@@ -328,3 +331,143 @@ def move(
         msg += f"Self has {length} elements, end_index: {end_index}"
         raise ValueError(msg)
     return moving_function(position=stack_element.position - start_index, n_elements=end_index - start_index)
+
+
+def bool_to_moving_function(is_rolled: bool) -> Union[pick, roll]:
+    """Map is_rolled (bool) to corresponding moving function."""
+    return roll if is_rolled else pick
+
+
+def reverse_endianness(
+    length: int,
+    stack_element: StackBaseElement = StackBaseElement(0),  # noqa: B008
+    rolling_option: bool = True,
+) -> Script:
+    """Reverse the endianness of a StackBaseElement of byte length `length`.
+
+    Args:
+        length (int): The byte length of stack_element.
+        stack_element (StackBaseElement): The stack element whose endianness should be reversed.
+            Defaults to `StackBaseElement(0)`.
+        rolling_option (bool): Whether stack_element should be rolled. Defaults to `True`.
+
+    """
+    out = move(stack_element, roll if rolling_option else pick)  # Move stack_element
+    out += Script.parse_string(" ".join(["OP_1 OP_SPLIT"] * (length - 1)))
+    out += Script.parse_string(" ".join(["OP_SWAP OP_CAT"] * (length - 1)))
+    return out
+
+
+def reverse_endianness_unknown_length(
+    max_length: int,
+    stack_element: StackBaseElement = StackBaseElement(0),  # noqa: B008
+    rolling_option: bool = True,
+) -> Script:
+    """Reverse the endianness of a StackBaseElement of byte length at most `max_length`.
```

Comment:
I think the naming may be a tiny bit misleading. In the end it is not really unknown, it is max_lenght, right?

Reply:
Discussed and modified.